src/components/bodhi/createTopic.js

```diff
@@ -262,14 +161,298 @@ class CreateTopic extends React.Component {
+          bettingStartBlock: block,
+        });
+        this.props.form.setFieldsValue({
+          bettingStartTime: localDate,
+        });
+        break;
+      }
+      case ID_BETTING_END_TIME: {
+        this.setState({
+          bettingEndBlock: block,
+        });
+        this.props.form.setFieldsValue({
+          bettingEndTime: localDate,
+        });
+        break;
+      }
+      case ID_RESULT_SETTING_START_TIME: {
+        this.setState({
+          resultSettingStartBlock: block,
+        });
+        this.props.form.setFieldsValue({
+          resultSettingStartTime: localDate,
+        });
+        break;
+      }
+      case ID_RESULT_SETTING_END_TIME: {
+        this.setState({
+          resultSettingEndBlock: block,
+        });
+        this.props.form.setFieldsValue({
+          resultSettingEndTime: localDate,
+        });
+        break;
+      }
+      default: {
+        throw new Error(`Unhandled id ${id}`);
+      }
+    }
+  }
+
+  validateTitleLength(rule, value, callback) {
+    let hexString = _.isUndefined(value) ? '' : value;
+
+    // Remove hex prefix for length validation
+    hexString = Web3Utils.toHex(hexString).slice(2);
+    if (hexString && hexString.length <= MAX_LEN_EVENTNAME_HEX) {
+      callback();
+    } else {
+      callback('Event name is too long.');
+    }
+  }
+
+  validateBettingEndTime(rule, value, callback) {
+    const bettingStartTime = this.props.form.getFieldValue(ID_BETTING_START_TIME);
+    if (_.isUndefined(bettingStartTime) || value.unix() <= bettingStartTime.unix()) {
+      callback('Must be greater than Betting Start Time');
+    } else {
+      callback();
+    }
+  }
+
+  validateResultSettingStartTime(rule, value, callback) {
+    const bettingEndTime = this.props.form.getFieldValue(ID_BETTING_END_TIME);
+    if (_.isUndefined(bettingEndTime) || value.unix() < bettingEndTime.unix()) {
+      callback('Must be greater than or equal to Betting End Time');
+    } else {
+      callback();
+    }
+  }
+
+  validateResultSettingEndTime(rule, value, callback) {
+    const resultSettingStartTime = this.props.form.getFieldValue(ID_RESULT_SETTING_START_TIME);
+    if (_.isUndefined(resultSettingStartTime) || value.unix() <= resultSettingStartTime.unix()) {
+      callback('Must be greater than Result Setting Start Time');
+    } else {
+      callback();
+    }
+  }
+
+  handleSubmit(evt) {
+    evt.preventDefault();
+
+    this.props.form.validateFieldsAndScroll((err, values) => {
+      if (!err) {
+        // Maps form variables to saga request variables
+        const {
+          centralizedOracle,
+          name,
+          results,
+          bettingStartTime,
+          bettingEndTime,
+          resultSettingStartTime,
+          resultSettingEndTime,
+        } = values;
+
+        this.props.onCreateTopic({
+          centralizedOracle,
+          name,
+          results,
+          bettingStartTime: bettingStartTime.utc().unix().toString(),
```

Comment:
toString() is not neccessary

Reply:
On the UI -> qweb3 encoding side, I decided to pass in all and any UINTs as a string (non-hex). This solves a couple of issues:

1. A string representation of a decimal number handles any range of number that is limited by JS's limit of 2^53.

2. Not having the need to convert it to a hex - on qweb3 side, I am purposefully forcing the user to use the `0x` prefix for any hex values, otherwise I am treating any strings as decimal representations. without forcing the hex prefix, ie. `'15231'` can either be `15231` in decimal or `86577` in hex. so I figured I would just pass in the decimal representation as a string to make things simpler.

Reply:
@dwalintukan 2^32 is bit enough for unix timestamp. no need for string or hex for now

In the backend side it will be stored as int anyway so $lt etc can be performed